### DISCUSSION TITLE
How can I filter a query on a subquery with max(value)?

### DISCUSSION THREAD

**Discussion by nik-lampe** [mikro-orm/mikro-orm#3377] [Q&A] [answered]
Hello,

I'm trying to wrap my head around this for a while now, I guess I just have to ask for help :D 

Let's assume I have the model design like this:

`create table entries (
    id    INT auto_increment
        primary key,
    value DOUBLE       not null,
    entry_date  DATE         not null,
    type  VARCHAR(255) not null
)`

The SQL query I want to do is roughly:

`SELECT * FROM entries e1 WHERE entry_date = (SELECT MAX(entry_date) AS max_date FROM entries e2 WHERE e2.type = e1.type)'

And I tried a lot, but I can't really grasp the concept of "registering via qb.withSubquery()" mentioned here:
https://mikro-orm.io/docs/query-builder#using-sub-queries

Any idea how I can do this?

I managed to do it with knex directly, but it's kinda complicated to map the result back to entities when there are joins involved (like in my real world use case)

**Comment by B4nan** [comment]
> And I tried a lot

So what have you tried? Show some code.

And check the QB tests to see how this can be done:

https://github.com/mikro-orm/mikro-orm/blob/master/tests/QueryBuilder.test.ts#L1913-L1936

**Reply by nik-lampe** [reply]
I didn't include the code yet, because it's a currently just a big mess of many different attempts, which handles a different datamodel than the simplified I mentioned above.

But I will try to recreate it and share it here later. 

Just thought that maybe I'm just missing an obvious solution for this. 😀

**Reply by nik-lampe** [reply]
Will also look into the QB tests, thank you!

**Reply by nik-lampe** [reply]
@B4nan 

Thanks! 

The tests gave me the right direction.

I wasn't aware that I can just drop the SQL query of the other querybuilder into my where clause like here:
https://github.com/mikro-orm/mikro-orm/blob/73a0ab35a6682034a8830e0f26409ccbb080ef63/tests/QueryBuilder.test.ts#L1928

Maybe this could be mentioned in the docs? 

This worked for me:
```
const knex = em.getKnex();

const qb2 = em
  .createQueryBuilder(Entry, "e2")
  .select("max(date) as maxDate")
  .where({ type: knex.ref("a.type") });

const qb1 = em
  .createQueryBuilder(Entry, "a")
  .select("*")
  .where(`date = (${qb2.getFormattedQuery()}) `);

const entries = await qb1.getResultList();
```

**Comment by nik-lampe** [comment, marked as answer]
This solved my needs:

```
const knex = em.getKnex();

const qb2 = em
  .createQueryBuilder(Entry, "e2")
  .select("max(date) as maxDate")
  .where({ type: knex.ref("a.type") });

const qb1 = em
  .createQueryBuilder(Entry, "a")
  .select("*")
  .where(`date = (${qb2.getFormattedQuery()}) `);

const entries = await qb1.getResultList();
```
